pennylane/plugins/expt_tensornet.py

```diff
@@ -162,15 +160,49 @@ def pre_apply(self):
         self.reset()
 
     def apply(self, operation, wires, par):
+        if operation == "QubitStateVector":
+            state = np.asarray(par[0], dtype=np.complex128)
+            if state.ndim == 1 and state.shape[0] == 2 ** self.num_wires:
+                self._state_node.tensor = np.reshape(state, [2] * self.num_wires)
+            else:
+                raise ValueError("State vector must be of length 2**wires.")
+            if wires is not None and wires != [] and list(wires) != list(range(self.num_wires)):
+                raise ValueError(
+                    "The expt.tensornet plugin can apply QubitStateVector only to all of the {} wires.".format(
+                        self.num_wires
+                    )
+                )
```

Comment:
Note that support for applying QubitStateVector to any subset of wires has been added in #346. It might be useful to turn that code into a function within `utils.py`, that can be used by any device if needed without code duplication.

Reply:
Agreed. Since this device is still just experimental, I'd like to leave that to a future PR